src/OAuthHttpClient.php

```diff
@@ -1,36 +1,37 @@
 <?php
 
+declare(strict_types=1);
+
 namespace BenjaminFavre\OAuthHttpClient;
 
-use BenjaminFavre\OAuthHttpClient\TokensCache\MemoryTokensCache;
-use BenjaminFavre\OAuthHttpClient\TokensCache\TokensCacheInterface;
 use BenjaminFavre\OAuthHttpClient\GrantType\GrantTypeInterface;
-use BenjaminFavre\OAuthHttpClient\ResponseChecker\ResponseCheckerInterface;
-use BenjaminFavre\OAuthHttpClient\ResponseChecker\StatusCode401ResponseChecker;
 use BenjaminFavre\OAuthHttpClient\GrantType\RefreshableGrantTypeInterface;
 use BenjaminFavre\OAuthHttpClient\RequestSigner\BearerHeaderRequestSigner;
 use BenjaminFavre\OAuthHttpClient\RequestSigner\RequestSignerInterface;
+use BenjaminFavre\OAuthHttpClient\ResponseChecker\ResponseCheckerInterface;
+use BenjaminFavre\OAuthHttpClient\ResponseChecker\StatusCode401ResponseChecker;
+use BenjaminFavre\OAuthHttpClient\TokensCache\MemoryTokensCache;
+use BenjaminFavre\OAuthHttpClient\TokensCache\TokensCacheInterface;
 use RuntimeException;
 use Symfony\Contracts\HttpClient\HttpClientInterface;
 use Symfony\Contracts\HttpClient\ResponseInterface;
 use Symfony\Contracts\HttpClient\ResponseStreamInterface;
 
-class OAuthHttpClient implements HttpClientInterface
+final class OAuthHttpClient implements HttpClientInterface
```

Comment:
Please notice that I have added `final` because of this Psalm error:

```
ERROR: UnsafeInstantiation - src/OAuthHttpClient.php:94:16 - Cannot safely instantiate class BenjaminFavre\OAuthHttpClient\OAuthHttpClient with "new static" as its constructor might change in child classes (see https://psalm.dev/229)
        return new static($this->client->withOptions($options), $this->grant);
```

This is effectively a BC break, so if you're not okay with that at this point, let me know :-)